### PR TITLE
Removed the space between the thead and tbody of the Table Component.

### LINKED ISSUE #24439: Table: there is an unexcepted gap between the thead and tbody elements
### Environment Information

- **Package version(s)**: v9.2.0
- **Browser and OS versions**: Edge (Chromium 104)

### Describe the issue:

There is an gap between the thead and tbody elements in the table component.
image

#### Actual behavior:

No gap/space should show between the table head and table body.

#### Expected behavior:

Gap/space between the table head and table body.

### If applicable, please provide a codepen repro:

The solution: just to add a CSS property onto the table element:
```css
border-collapse: collapse;
```

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? No

Requested priority: Low

Products/sites affected: https://react.fluentui.dev/?path=/docs/preview-components-table--default

### PR DESCRIPTION
**Pull request by KBoeh**
## Current Behavior

This fix affects the react-table package. There is a space between the thead and tbody elements in the table component.
Before

## New Behavior

I added 'border-collapse: collapse' property in order to remove the space between the thead and tbody.
After

## Related Issue(s)

[](https://github.com/microsoft/fluentui/issues/24439)

Fixes #24439

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.656 kB`<br />`51.96 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `33.359 kB`<br />`11.004 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |   `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/7f8bad094dadcdb671d5746848e96b5bd4b5791a'>7f8bad094dadcdb671d5746848e96b5bd4b5791a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/Avatar-mount_base.html">1303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/Avatar-mount.html">1465</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/Button-mount_base.html">1115</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/Button-mount.html">1117</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProvider-mount_base.html">1625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProvider-mount.html">1689</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-mount_base.html">638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-mount.html">637</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">581</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">602</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">619</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/MakeStyles-mount_base.html">1882</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/MakeStyles-mount.html">1895</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/SpinButton-mount_base.html">2538</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24748/perf-test-react-components/SpinButton-mount.html">2505</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-table/src/components/Table/useTableStyles.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export const tableClassNames: SlotClassNames<TableSlots> = {
 const useStyles = makeStyles({
   root: {
     width: '100%',
+    'border-collapse': 'collapse',
```
Comment (layershifter) [inline packages/react-components/react-table/src/components/Table/useTableStyles.ts:16]:
```suggestion
    borderCollapse: 'collapse',
```

**Comment by ling1726** [comment]
Thanks for the contribution @KBoeh 🙏. Unfortunately, there are big changes upcoming in #24762 that are going to be targeted for the next preview release of the table component. The `border-collapse` styling will be there

**Comment by KBoeh** [comment]
No problem. This was my first ever open-source PR and I earned allot. Maybe next time.

> Thanks for the contribution @KBoeh 🙏. Unfortunately, there are big changes upcoming in #24762 that are going to be targeted for the next preview release of the table component. The `border-collapse` styling will be there

**Comment by KBoeh** [comment]
> Thanks for the contribution @KBoeh 🙏. Unfortunately, there are big changes upcoming in #24762 that are going to be targeted for the next preview release of the table component. The `border-collapse` styling will be there

No problem. This was my first ever open-source PR and I learned allot. Maybe next time!

**Pull request closed without merging**